src/courtroom.cpp

```diff
@@ -1878,11 +1882,20 @@ void Courtroom::handle_chatmessage(QStringList *p_contents)
     ic_chatlog_history.removeFirst();
   }
 
-  append_ic_text(m_chatmessage[MESSAGE], f_showname, "", m_chatmessage[TEXT_COLOR].toInt());
-
   QString f_char = m_chatmessage[CHAR_NAME];
   QString f_custom_theme = ao_app->get_char_shouts(f_char);
 
+  // Load the colors in case it's using a custom chatbox with custom colors.
+  // Or reload the default ones in case it's not using custom colors
+  color_rgb_list.clear();
+  for (int c = 0; c < max_colors; ++c) {
+    QColor color = ao_app->get_chat_color("c" + QString::number(c), f_char);
```

Comment:
Wait wouldn't this be kind of expensive, it's a file lookup every time a message is added to the IC logs right?? Shouldn't this be moved to the widget function?

Reply:
In fact I think custom chatbox colors shouldn't be supported in the logs at all, make the logs stick to one color foundation instead

Reply:
The colors actually have to be loaded in every time a new message comes in, because the colors might be different in the case of a misc folder